File: scripts/temp_read_pdfs.py
import sys
import os

try:
    import pypdf
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "pypdf"])
    import pypdf

def extract_pdf(pdf_path, txt_path):
    print(f"Extracting {pdf_path} to {txt_path}...")
    try:
        with open(pdf_path, 'rb') as f:
            reader = pypdf.PdfReader(f)
            text = ''
            for page in reader.pages:
                extracted = page.extract_text()
                if extracted:
                    text += extracted + '\n'
            
            os.makedirs(os.path.dirname(txt_path), exist_ok=True)
            with open(txt_path, 'w', encoding='utf-8') as out:
                out.write(text)
        print(f"Successfully extracted {pdf_path}")
    except Exception as e:
        print(f"Error extracting {pdf_path}: {e}")

btech_pdf = r"d:\notebook lm\syllabus\BTech_Syllabus_detailed.pdf"
mbbs_pdf = r"d:\notebook lm\syllabus\Syllabus - MBBS.pdf"
btech_txt = r"d:\notebook lm\data\btech_extracted.txt"
mbbs_txt = r"d:\notebook lm\data\mbbs_extracted.txt"

extract_pdf(btech_pdf, btech_txt)
extract_pdf(mbbs_pdf, mbbs_txt)

File: scripts/temp_read_pdfs.js
const fs = require('fs');
const pdf = require('pdf-parse');
const path = require('path');

async function extractPdf(pdfPath, txtPath) {
    try {
        console.log(`Extracting ${pdfPath}...`);
        let dataBuffer = fs.readFileSync(pdfPath);
        let data = await pdf(dataBuffer);
        
        const dir = path.dirname(txtPath);
        if (!fs.existsSync(dir)) {
            fs.mkdirSync(dir, { recursive: true });
        }
        
        fs.writeFileSync(txtPath, data.text, 'utf-8');
        console.log(`Successfully extracted to ${txtPath}`);
    } catch (err) {
        console.error(`Error extracting ${pdfPath}:`, err.message);
    }
}

const btechPdf = "d:\\notebook lm\\syllabus\\BTech_Syllabus_detailed.pdf";
const mbbsPdf = "d:\\notebook lm\\syllabus\\Syllabus - MBBS.pdf";
const btechTxt = "d:\\notebook lm\\data\\btech_extracted.txt";
const mbbsTxt = "d:\\notebook lm\\data\\mbbs_extracted.txt";

async function run() {
    await extractPdf(btechPdf, btechTxt);
    await extractPdf(mbbsPdf, mbbsTxt);
}

run();
